### PR TITLE
Improve error message for entity not found in flytekit cli

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue
NA

## Why are the changes needed?
error message is not clear for users

## What changes were proposed in this pull request?
Improve error message for entity not found in flytekit cli

## How was this patch tested?
```bash
pyflyte -vv run --remote flyte-example/getting_started.py wf1
```

### Screenshots
Before:
Screenshot 2024-08-26 at 2 21 15 PM

After:
Screenshot 2024-08-26 at 1 57 45 PM

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
NA

## Docs link
NA

### CODE REVIEW

File: flytekit/exceptions/user.py [hunk 1]
```diff
@@ -64,6 +64,15 @@ def __init__(self, path: str):
         super(FlyteDataNotFoundException, self).__init__(path, "File not found")
 
 
+class FlyteEntityNotFoundException(FlyteValueException):
+    def __init__(self, module_name: str, entity_name: str):
+        self._module_name = module_name
+        self._entity_name = entity_name
+
+    def __str__(self):
+        return f"Entity '{self._entity_name}' not found in module '{self._module_name}'"
```
Comment (thomasjpfan) [inline flytekit/exceptions/user.py:73]: For a new flyte user, I do not think we explain what an "entity" is. I'm looking over the docs and it's hard to find a definition.

Comment (pingsutw) [reply flytekit/exceptions/user.py:73]: how about "task" or "workflow"?

Comment (pingsutw) [reply flytekit/exceptions/user.py:73]: "Task/Workflow '{self._entity_name}' not found in module"

Comment (thomasjpfan) [reply flytekit/exceptions/user.py:73]: I'm +1 on `Task/Workflow`.

**Review by eapolinario — APPROVED** [review]

**Pull request merged**